runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -631,7 +632,10 @@ protected void rollbackStreamUnsafe(ISMRStream stream, long rollbackVersion) {
 
         List<SMREntry> entries = stream.current();
 
-        while (!entries.isEmpty()) {
+        while (stream.pos() > rollbackVersion) {
```

Comment:
If the SMREntry is a no-op we can skip calling applyUndo, likewise we can skip the apply on stream forward?